scripts/mirror.test.js

```diff
@@ -0,0 +1,141 @@
+
+describe('mirror', function () {
+  const match = (source_browser, source_version, dest_browser) => {
+    const source_release = browsers[source_browser].releases[source_version];
+    return getMatchingBrowserVersion(dest_browser, source_release);
+  };
+
+  it('chrome_android', function () {
+    assert.equal(match('chrome', '1', 'chrome_android'), '18');
+    assert.equal(match('chrome', '2', 'chrome_android'), '18');
+    assert.equal(match('chrome', '17', 'chrome_android'), '18');
+    assert.equal(match('chrome', '18', 'chrome_android'), '18');
+    assert.equal(match('chrome', '19', 'chrome_android'), '25');
+    assert.equal(match('chrome', '24', 'chrome_android'), '25');
+    assert.equal(match('chrome', '25', 'chrome_android'), '25');
+    assert.equal(match('chrome', '26', 'chrome_android'), '26');
+  });
+
+  it('edge', function () {
+    assert.equal(match('chrome', '1', 'edge'), false);
+    assert.equal(match('chrome', '27', 'edge'), false);
+    assert.equal(match('chrome', '28', 'edge'), '79');
+    assert.equal(match('chrome', '79', 'edge'), '79');
+    assert.equal(match('chrome', '80', 'edge'), '80');
+  });
+
+  it('firefox_android', function () {
+    assert.equal(match('firefox', '1', 'firefox_android'), '4');
+    assert.equal(match('firefox', '2', 'firefox_android'), '4');
+    assert.equal(match('firefox', '3', 'firefox_android'), '4');
+    assert.equal(match('firefox', '4', 'firefox_android'), '4');
+    assert.equal(match('firefox', '68', 'firefox_android'), '68');
+    assert.equal(match('firefox', '69', 'firefox_android'), '79');
+    assert.equal(match('firefox', '79', 'firefox_android'), '79');
+    assert.equal(match('firefox', '80', 'firefox_android'), '80');
+  });
+
+  it('opera', function () {
+    assert.equal(match('chrome', '1', 'opera'), '15');
+    assert.equal(match('chrome', '27', 'opera'), '15');
+    assert.equal(match('chrome', '28', 'opera'), '15');
+    assert.equal(match('chrome', '29', 'opera'), '16');
+    assert.equal(match('chrome', '30', 'opera'), '17');
+  });
+
+  it('opera_android', function () {
+    assert.equal(match('chrome_android', '18', 'opera_android'), '14');
+    assert.equal(match('chrome_android', '26', 'opera_android'), '14');
+    assert.equal(match('chrome_android', '27', 'opera_android'), '14'); // wrong, should be 15
+    assert.equal(match('chrome_android', '28', 'opera_android'), '15');
+    assert.equal(match('chrome_android', '29', 'opera_android'), '16');
+    assert.equal(match('chrome_android', '30', 'opera_android'), '18');
+  });
+
+  it('safari_ios', function () {
+    assert.equal(match('safari', '1', 'safari_ios'), '1');
+    assert.equal(match('safari', '1.1', 'safari_ios'), '1');
+    assert.equal(match('safari', '1.2', 'safari_ios'), '1');
+    assert.equal(match('safari', '1.3', 'safari_ios'), '1');
+    assert.equal(match('safari', '2', 'safari_ios'), '1');
+    assert.equal(match('safari', '3', 'safari_ios'), '2'); // wrong, should be 1
```

Comment:
Actually I'm not sure about this. Safari 3 used WebKit 523.10, and Safari iOS 1 used WebKit 522.11, while iOS 2 used 525.18. So the mapping makes sense. But what Safari version did iOS 1 claim to have in the UA string?